src/Tgstation.Server.Host/ServerFactory.cs

```diff
@@ -49,7 +49,11 @@ public async Task<IServer> CreateServer(string[] args, string updatePath, Cancel
 
 			var basePath = IOManager.ResolvePath();
 			IHostBuilder CreateDefaultBuilder() => Microsoft.Extensions.Hosting.Host.CreateDefaultBuilder(args)
-				.ConfigureAppConfiguration((context, configuration) => configuration.SetBasePath(basePath));
+				.ConfigureAppConfiguration((context, builder) => {
+					builder.AddYamlFile("appsettings.yml", optional: true, reloadOnChange: true)
+					.AddYamlFile($"appsettings.{context.HostingEnvironment.EnvironmentName}.yml", optional: true, reloadOnChange: true);
```

Comment:
To fix it we'll have to reorganize the builder's sources after creating them

```suggestion
					builder.AddYamlFile("appsettings.yml", optional: true, reloadOnChange: true)
						.AddYamlFile($"appsettings.{context.HostingEnvironment.EnvironmentName}.yml", optional: true, reloadOnChange: true);

					// reorganize the builder so our yaml configs don't override the env/cmdline configs
					// values obtained via debugger
					var envConfig = builder.Sources[3];
					var cmdLineConfig = builder.Sources[4];
					var baseYmlConfig = builder.Sources[5];
					var environmentYmlConfig = builder.Sources[6];
					builder.Sources[3] = baseYmlConfig;
					builder.Sources[4] = environmentYmlConfig;
					builder.Sources[5] = envConfig;
					builder.Sources[6] = cmdLineConfig;
```